flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -521,6 +521,19 @@ private static IconCompat getIconFromSource(Context context, String iconPath, Ic
                     throw new RuntimeException(e);
                 }
                 break;
+            case ByteArray:
+                byte[] byteArray;
```

Comment:
Would you be able to refactor the logic around assigning the value of `byteArray`? It's a repeat of what was in `getBitmapFromSource`.

Reply:
done